### PR TITLE
spiffetls modes scaffolding

### PR DESCRIPTION
**Pull request by azdagron**

### CODE REVIEW

File: v2/spiffetls/tlsconfig/config.go [hunk 1]
```diff
@@ -46,19 +46,21 @@ func HookMTLSClientConfig(config *tls.Config, svid x509svid.Source, bundle x509b
...
-	HookMTLSWebClientConfig(config, svid)
+	HookMTLSWebClientConfig(config, svid, roots)
 	return config
 }
 
 // HookMTLSWebClientConfig sets up the TLS configuration to present an
-// X509-SVID to the server and verified the server certificate using system
-// roots.
-func HookMTLSWebClientConfig(config *tls.Config, svid x509svid.Source) {
+// X509-SVID to the server and verified the server certificate using the
```
Comment (amartinezfayo) [inline v2/spiffetls/tlsconfig/config.go:58]:
```suggestion
// X509-SVID to the server and verifies the server certificate using the
```

File: v2/spiffetls/mode.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+// authenticate the server X509-SVID are obtained via a new Workload API X.509
+// source created with the provided source options. The authorizer is used to
+// authorize the server X509-SVID.
+func MTLSClientWithSourceOptions(options []workloadapi.X509SourceOption, authorizer tlsconfig.Authorizer) DialMode {
+	panic("not implemented")
+}
+
+// MTLSClientWithConfig configures the dialing for mutually authenticated TLS
+// (mTLS). The client X509-SVID and the X.509 bundles used to authenticate the
+// server X509-SVID are obtained via the provided sources. The sources must
```
Comment (amartinezfayo) [inline v2/spiffetls/mode.go:76]:
```suggestion
// server X509-SVID are obtained via the provided X509-SVID source. The source must
```

File: v2/spiffetls/mode.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+// authenticated TLS (mTLS).  The client X509-SVID is obtained via a new
+// Workload API X.509 source created with the provided source options. The
+// roots (or the system roots if nil) are used to authenticate the server
+// certificate).
+func MTLSWebClientWithSourceOptions(options []workloadapi.X509SourceOption, roots *x509.CertPool) DialMode {
+	panic("not implemented")
+}
+
+// MTLSWebClientWithConfig configures the dialing for mutually authenticated
+// TLS (mTLS). The client X509-SVID is obtained via the provided source. The
```
Comment (amartinezfayo) [inline v2/spiffetls/mode.go:109]:
```suggestion
// TLS (mTLS). The client X509-SVID is obtained via the provided X509-SVID source. The
```

File: v2/spiffetls/tlsconfig/config.go [hunk 4]
```diff
@@ -46,19 +46,21 @@ func HookMTLSClientConfig(config *tls.Config, svid x509svid.Source, bundle x509b
 }
 
 // MTLSWebClientConfig returns a TLS configuration which presents an X509-SVID
-// to the server and verifies the server certificate using system roots.
-func MTLSWebClientConfig(svid x509svid.Source) *tls.Config {
+// to the server and verifies the server certificate using provided roots (or
+// the system roots if nil)
+func MTLSWebClientConfig(svid x509svid.Source, roots *x509.CertPool) *tls.Config {
```
Comment (amartinezfayo) [inline v2/spiffetls/tlsconfig/config.go:51]: We may explicit in the comment what's the role of the x509svid.Source?

Comment (azdagron) [reply v2/spiffetls/tlsconfig/config.go:51]: If we want to do that, let's tackle that in a follow up PR. Doing so will require comment changes on all of the `tlsconfig` methods for consistency.

**Pull request merged**